addons/sale/tests/test_sale_order_discount.py

```diff
@@ -83,3 +84,22 @@ def test_sol_discount(self):
             all(line.discount == 50 for line in self.sale_order.order_line)
         )
         self.assertAlmostEqual(self.sale_order.amount_untaxed, so_amount*0.5)
+
+    def test_sol_discount_removal(self):
+        so_amount = self.sale_order.amount_untaxed
+        self.wizard.write({'discount_percentage': 0.5, 'discount_type': 'sol_discount'})
+        self.wizard.action_apply_discount()
+
+        self.wizard.write({'discount_percentage': 0, 'discount_type': 'sol_discount'})
+        self.wizard.action_apply_discount()
+
+        self.assertFalse(len(self.sale_order.order_line.filtered('discount')))
```

Comment:
An empty recordset is falsy
```suggestion
        self.assertFalse(self.sale_order.order_line.filtered('discount'))
```